online/src/app/edit-product-form/edit-product-form.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Product } from '../products';
+import { ProductService } from '../services/product.service';
+import { Location } from '@angular/common'
+@Component({
+  selector: 'app-edit-product-form',
+  templateUrl: './edit-product-form.component.html',
+  styleUrls: ['./edit-product-form.component.scss']
+})
+export class EditProductFormComponent implements OnInit {
+  editForm: FormGroup | undefined;
+  product: Product | undefined;
+  constructor(private productService: ProductService, private route: ActivatedRoute, private formBuilder: FormBuilder,
+    private location: Location) { }
+
+  ngOnInit(): void {
+    const id = this.route.snapshot.params['id'];
+    this.productService.getProductById(id).subscribe((item: Product) => {
```

Comment:
You'll only want this once so I suggest you add a pipe with a take(1) instruction, such that it automatically unsubscribes as well, making your code less prone to memory leaks